### PR TITLE
Eei story 2b

### PR DESCRIPTION
**Pull request by Kinshuk1993**
This is to demonstrate resilience to poor DB performance
- Develop throttled ACE bar for DB Writer
  - Change code for DB Writer to deliberately only pull 1 message per second from MQ
- Create a test script to load test the POST call
- Doc how to run the load script, watch the MQ queue grow, stop the load script, watch the MQ queue shrink

### CODE REVIEW

File: products/bash/1-click-pre-validation.sh [hunk 1]
```diff
@@ -74,6 +74,11 @@ if [[ -z "${namespace// }" ]]; then
   missingParams="true"
 fi
 
+if [[ -z "${namespace// }" ]]; then
```
Comment (laurensks) [inline products/bash/1-click-pre-validation.sh:77]: You have a duplicate check for the namespace here 
image

Comment (Kinshuk1993) [reply products/bash/1-click-pre-validation.sh]: Done

**Comment by Kinshuk1993** [comment]
New DB WRITER code to include changes from main:
```
public void evaluate(MbMessageAssembly inAssembly) throws MbException {
	//	Uncomment to debug
	//	MbElement root = inAssembly.getMessage().getRootElement();
	//	System.out.println("root: " + root);
	//	printElement(root, "");

	MbOutputTerminal out = getOutputTerminal("out");
	MbOutputTerminal alt = getOutputTerminal("alternate");

	MbMessage inMessage = inAssembly.getMessage();
	MbMessageAssembly outAssembly = null;
	try {
		// create new message as a copy of the input
		MbMessage outMessage = new MbMessage(inMessage);
		outAssembly = new MbMessageAssembly(inAssembly, outMessage);
		// ----------------------------------------------------------
		// Add user code below
		try        
		{
			System.out.println("Rate throttling for one second");
		    Thread.sleep(1000);
		} 
		catch(InterruptedException ex) 
		{
		    Thread.currentThread().interrupt();
		}
		System.out.println("Getting message from MQ queue to write in database");
		// JSON/Data is the path JSON format of the MQ message
		MbElement DataRoot = outMessage.getRootElement().getLastChild().getFirstElementByPath("/JSON/Data");
		// Obtain a java.sql.Connection using a JDBC Type4 datasource
        // This example uses a Policy of type JDBCProviders called "MyJDBCPolicy" in Policy Project "MyPolicies"  
        Connection conn = getJDBCType4Connection("{DefaultPolicies}:PostgresqlPolicy", JDBC_TransactionType.MB_TRANSACTION_AUTO);
        // Example of using the Connection to create a java.sql.Statement 
        //Statement stmt = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);	

        Object name = DataRoot.getFirstElementByPath("name").getValue();
        Object email = DataRoot.getFirstElementByPath("email").getValue();
		Object age = DataRoot.getFirstElementByPath("age").getValue();
		Object address = DataRoot.getFirstElementByPath("address").getValue();
		Object usState = DataRoot.getFirstElementByPath("usState").getValue();
		Object licensePlate = DataRoot.getFirstElementByPath("licensePlate").getValue();
		Object descriptionOfDamage = DataRoot.getFirstElementByPath("descriptionOfDamage").getValue();
		Object quoteid = DataRoot.getFirstElementByPath("quoteid").getValue();
		Object source = "Mobile";
		Object claimStatus = 1;
		
		// checking and converting if age is string, age has to be an integer
        if ( age != null ) {
        	try {
        		age= Integer.valueOf(age.toString());
        	}
        	catch ( NumberFormatException e) {
        		age=null;
        	}
        }

        // Constructing the SQL Query
        String sql = "INSERT INTO QUOTES(Name,EMail,Age,Address,USState,LicensePlate,descriptionOfDamage,QuoteID, Source, ClaimStatus) "
        		+ "VALUES(?, ?, ?, ?, ? ,? ,?, ?, ?, ?) RETURNING *";
        
        PreparedStatement pstmt = conn.prepareStatement(sql);
        pstmt.setObject(1, name);
        pstmt.setObject(2, email);
        pstmt.setObject(3, age);
        pstmt.setObject(4, address);
        pstmt.setObject(5, usState);
        pstmt.setObject(6, licensePlate);
        pstmt.setObject(7, descriptionOfDamage);
        pstmt.setObject(8, quoteid);
        pstmt.setObject(9, source);
        pstmt.setObject(10, claimStatus);
        
        System.out.println("SQL String=" + sql);
        // executing the above query
        ResultSet rs = pstmt.executeQuery();       
        
		// End of user code
		// ----------------------------------------------------------
	} catch (MbException e) {
		// Re-throw to allow Broker handling of MbException
		e.printStackTrace();
	} catch (RuntimeException e) {
		// Re-throw to allow Broker handling of RuntimeException
		e.printStackTrace();
	} catch (Exception e) {
		e.printStackTrace();
	}
	// The following should only be changed
	// if not propagating message to the 'out' terminal
	out.propagate(outAssembly);

}
```

File: EventEnabledInsurance/post-load-test-readme.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- The MQ Queue `Quote` would initially be empty.
+<br /><br />
+
+# Running the test script
+- Run the [load testing script](post-load-test.sh) with `NAMESPACE` parameter ([exported above](post-load-test-readme.md#prerequisites)) and the desired number of POST calls to be made (`TARGET_POST_CALLS` - if any value other than the default is required):
+    ```
+    export TARGET_POST_CALLS = <TARGET_POST_CALLS>
+    ./post-load-test.sh -n $NAMESPACE -c $TARGET_POST_CALLS
+    ```
+- If `-n` not provided the script will default to namespace `cp4i`.
```
Comment (tim-lo) [inline EventEnabledInsurance/post-load-test-readme.md:39]:
```suggestion
- If `-n` is not provided the script will default to namespace `cp4i`.
```

File: EventEnabledInsurance/post-load-test-readme.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    At the end of the script it will display:
+    - Calls made in a second
+    - Calls failed (if any) 
+- If `-c` is not specified, the number of POST calls will default to `100`.
+<br /><br />
+
+# Checking the Queue
+- Assuming that the Integration server `ace-db-writer-int-srv-eei` is running fine, you should be able to see messages appearing disappearing from the `Quote` Queue in MQ Console on pressing the refresh icon. These messages are being added in the Postgres database table `db_cp4i1_sor_eei` by `ace-db-writer-int-srv-eei`.
```
Comment (tim-lo) [inline EventEnabledInsurance/post-load-test-readme.md:47]:
```suggestion
- Assuming that the Integration server `ace-db-writer-int-srv-eei` is running fine, you should be able to see messages appearing and disappearing from the `Quote` Queue in MQ Console on pressing the refresh icon. These messages are being added in the Postgres database table `db_cp4i1_sor_eei` by `ace-db-writer-int-srv-eei`.
```

Comment (tim-lo) [reply EventEnabledInsurance/post-load-test-readme.md:47]: also the name fo the database would depend on the namespace name, minor detail but should probably doc that

Comment (Kinshuk1993) [reply EventEnabledInsurance/post-load-test-readme.md]: Added a note for that in the readme.

**Review by tim-lo — APPROVED** [review]
lgtm :+1:

**Pull request merged**